### ISSUE TITLE
The props file for visual studio should not change the value of ExecutablePath

### ISSUE THREAD

**Issue by JolyLoic** [conan-io/conan#1606]
Hello,

We still have issues with the resolution of #998 or #1052, and it prevents us from upgrading to a recent version of Conan. But I don't think the problem is `$(ExecutablePath)`or `%(ExecutablePath)`or whatever, it's that I don't see any reason for Conan to change this property.

The description of this property explains that it contains the path of the programs **used during compilation**. You can find this description in your file: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\1036\directories.xml` or `C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\1033\directories.xml` or something similar depending on your version of visual studio.

```xml
  <StringListProperty Name="ExecutablePath" DisplayName="Executable Directories"
                      Description="Path to use when searching for executable files while building a VC++ project.  Corresponds to environment variable PATH."
                      F1Keyword="VC.Project.VCDirectories.ExecutablePath"
                      Category="VC++" Separator=";" Subtype="folder" />

```

So in a nutshell, I don't think there is any need to change the value of this variable, and changing it seems to cause some issues, because on our environment, it replaces its value instead of adding to it, and therefore MsBuild can no longer find `cl.exe`. I would suggest just removing this line from the props file.

By the way, if you can tell why it was added in the first place, maybe I can find another variable more appropriate to perform the initial operation?

Thank you

**Comment by memsharded** [comment]
The truth is that I don't know. This generator has been more a community thing, if you do the git blame of the source file, you will see most of it comes from external contributors.

It seems you are right, maybe conan shouldn't modify this variable. I kindly request the feedback of @claasd, and @Xaltotun regarding this issue.

**Comment by memsharded** [comment]
Also, please, the help of @youcefl and @davidkjaer would be appreciated.

Can we leave the ``ExecutablePath`` unmodified?

**Comment by claasd** [comment]
I don't know what the idea behind settings this variable was. Maybe, if you install e.g. OpenSSL via conan, you would have the openSSL.exe available directly for Build steps without needing to import it.

One way to allow for both would be to move the definition to a `ConanExecutablePath` variable. This would not be used at all by default, but if you need it to be set you could overwrite the ExecutablePath in your own project.

Proposal: Move the content to a `ConanExecutablePaths` variable and leave `ExecutablePath` umodified.

**Comment by claasd** [comment]
@JolyLoic Where in your project file do you import the conan.pops? I remember that I had to move them further down in my vcxproj files to work properly. We needed to place them AFTER the element
`<PropertyGroup Label="UserMacros" />`. Before that, there are some imports to microsoft props files which seem to set some of the predefined variables variables. 

Can you please check that for your projects?

**Comment by memsharded** [comment]
> e.g. OpenSSL via conan, you would have the openSSL.exe available directly for Build steps without needing to import it.

Yes, might be, but right now I cannot think of a real use case, or some user requesting this.

Agree with your proposal. Lets try this, we'll get further feedback once released, I guess :) Thanks very much for your help!

**Comment by memsharded** [comment]
And here we go: https://github.com/conan-io/conan/pull/1608

**Comment by lasote** [comment]
Merged. Will be available in the next release. 
I don't know if the following use case exists, but probably makes more sense to fill the conan executable path variable with the `self.deps_env_vars.PATH` variable too. Just to take into account if any user needs it.

**Comment by JolyLoic** [comment]
@claasd It's true it is included close to the top of our projects, but it used to work before (and will hopefully start working again with this new change).
@lasote @memsharded Thant you for this change!

**Comment by lasote** [comment]
We will keep the issue open until we release. Thanks!

**Comment by memsharded** [comment]
@JolyLoic This has been released in 0.26, please update and try

**Comment by Arenoros** [comment]
I have a reason why this variable must be set. 
I use the package winflexbison, and its targets extension for vs which requires an executable from this package.
Disabling this option prevents us from embedding the build dependencies into VS the same way we do for cmake

**Comment by memsharded** [comment]
Hi @Arenoros 

This issue is very old, like 4 years ago. We have new, better generators for Visual, like ``MSBuildDeps`` and ``MSBuildToolchain``, which will survive in Conan 2.0 (already in alpha), and ``msbuild`` will be removed. I encourage having a look, and if there is something you would like to comment or report on them, please open a new issue. Thanks!
